### PR TITLE
Miscellaneous warnings and errors 

### PR DESCRIPTION
**Pull request by abidlabs**
Just a few improvements to our warnings and errors:

* Raises a warning in the `File` component or `UploadButton` component if someone passes in a `file_types` parameter along with `file_count=="dir"` (closes: #3077) 
* Raises a `gr.Error` instead of a regular Python error when you use `gr.Interface.load()` to load a model and there's an error querying the HF API to make it more easily readable (closes: #3194), looks like this:

image

* Improves error messages when returning incorrect number of values from an event handler (closes: #3082)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -53,6 +53,8 @@ No changes to highlight.
 
 ## Full Changelog:
 * Fix demos page css and add close demos button by [@aliabd](https://github.com/aliabd) in [PR 3151](https://github.com/gradio-app/gradio/pull/3151)
+* Better warnings and error messages (e.g. when there is a mismatch between the number
```
Comment (freddyaboulton) [inline CHANGELOG.md:56]: Nit: Mention the other issues we're closing as well

File: test/test_blocks.py [hunk 2]
```diff
@@ -531,6 +533,15 @@ def generic_update():
             }
             assert output["data"][1] == {"__type__": "update", "mode": "dynamic"}
 
+    def test_error_raised_if_num_outputs_mismatch(self):
+        with gr.Blocks() as demo:
+            textbox1 = gr.Textbox()
+            textbox2 = gr.Textbox()
+            button = gr.Button()
+            button.click(lambda x: x, textbox1, [textbox1, textbox2])
+        with pytest.raises(ValueError):
```
Comment (freddyaboulton) [inline test/test_blocks.py:542]: nit: I would `match=` against the expected error message here but maybe I'm just too paranoid 😂

**Review by freddyaboulton — APPROVED** [review]
Looks great to me @abidlabs !! Thanks

**Comment by abidlabs** [comment]
Thanks for the review @freddyaboulton! Agree with the suggestions and have made the updates

**Pull request merged**